osm2gtfs/tests/creators/tests_creators_abstract.py

```diff
@@ -0,0 +1,322 @@
+        self.data_dir = os.path.realpath(os.path.join(current_dir, "../../../data/"))
+        self.fixture_dir = os.path.join(os.path.realpath(current_dir), "fixtures/" + self.selector)
+        self.output_file = os.path.realpath(os.path.join(
+            self.data_dir, self.selector + "_tests.zip"))
+
+        # Define fixtures files (need to be added to the fixtures directory for each provider)
+        self.reference_gtfs = os.path.join(self.fixture_dir, self.selector + "_gtfs.zip.ref")
+        self.mocked_overpass_routes = os.path.join(self.fixture_dir, "overpass-routes.xml")
+        self.mocked_overpass_stops = os.path.join(self.fixture_dir, "overpass-stops.xml")
+
+        # Load configuration
+        self.config = Configuration(TestCreatorsArgs(self.config_file, self.selector))
+
+        # Eventually configuration options can be overridden here
+        #   Example: Use local timetable.json
+        #   self.config.data['schedule_source'] = os.path.join(self.fixture_dir, "timetable.json")
+
+    def test_refresh_routes_cache(self):
+        data = OsmConnector(self.config)
+        cache_file = os.path.join(self.data_dir, self.selector + "-routes.pkl")
+        mocked_overpass_data_file = self.mocked_overpass_routes
+        if os.path.isfile(cache_file):
+            os.remove(cache_file)
+        with patch("osm2gtfs.core.osm_connector.OsmConnector._query_routes") as mocked1:
+            overpass_xml = open(mocked_overpass_data_file, mode='r').read()
+            api = overpy.Overpass()
+            mocked1.return_value = api.parse_xml(overpass_xml)
+            data.get_routes(refresh=True)
+        self.assertTrue(os.path.isfile(cache_file), 'The routes cache file creation failed')
+        cache = Cache()
+        routes = cache.read_data(self.selector + "-routes")
+        self.assertEqual(len(routes), self.count_routes, 'Wrong count of routes in the cache file')
+
+    def test_refresh_stops_cache(self):
+        data = OsmConnector(self.config)
+        cache_file = os.path.join(self.data_dir, self.selector + "-stops.pkl")
+        mocked_overpass_data_file = self.mocked_overpass_stops
+        if os.path.isfile(cache_file):
+            os.remove(cache_file)
+        with patch("osm2gtfs.core.osm_connector.OsmConnector._query_stops") as mocked1:
+            overpass_xml = open(mocked_overpass_data_file, mode='r').read()
+            api = overpy.Overpass()
+            mocked1.return_value = api.parse_xml(overpass_xml)
+            data.get_stops(refresh=True)
+        self.assertTrue(os.path.isfile(cache_file), 'The stops cache file creation failed')
+        cache = Cache()
+        stops = cache.read_data(self.selector + "-stops")
+        amount_of_stops = len(stops['regular']) + len(stops['stations'])
+        self.assertEqual(
+            # amount_of_stops, self.count_stops,  # WORKS FOR ACCRA
+            amount_of_stops, self.count_stops + self.count_stations,  # WORKS FOR MANAGUA
```

Comment:
Here we have an incompatibility between Accra and Managua and I would like to ask for your help on this, @prhod and @nlehuby.

For some reason the GTFS for Managua counts stops and stations (stop_areas), whereas Accra only counts the stops and not the stations. However in line [99](https://github.com/xamanu/osm2gtfs/blob/db43c073c0fc73bcf90f973237b17ccfe9402d05/osm2gtfs/tests/creators/tests_creators_abstract.py#L99) explicitly stops and stations are added up. So I guess Managua is following the correct behaviour. But I'm really not sure on this one and how to fix it for Accra.

Do you have an idea?

Reply:
Because of some other changes, the code moved: The important part is now at line 140-[144](https://github.com/xamanu/osm2gtfs/blob/6ebfa85ec86cdc7531daac1c33876422079ed42c/osm2gtfs/tests/creators/tests_creators_abstract.py#L144)